hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsModule.java

```diff
@@ -253,4 +261,36 @@ static Supplier<CallLocalEvmTxProcessor> provideCallLocalEvmTxProcessorFactory(
         return () -> new CallLocalEvmTxProcessor(
                 codeCache, livePricesSource, dynamicProperties, gasCalculator, mcps, ccps, aliasManager);
     }
+
+    @Provides
+    @Singleton
+    @Named("HederaPrecompiledContractsDetector")
+    static Predicate<Address> provideHederaPrecompiledContractsDetector() {
+        // all addresses between 0-750 (inclusive) are treated as system accounts
+        // from the perspective of the EVM when executing Call, Balance, and SelfDestruct operations
+        return address -> address.numberOfLeadingZeroBytes() >= 18
+                && (Integer.compareUnsigned(address.getInt(16), 359) == 0
+                        || Integer.compareUnsigned(address.getInt(16), 360) == 0
+                        || Integer.compareUnsigned(address.getInt(16), 361) == 0);
+    }
+
+    @Provides
+    @Singleton
+    @Named("HederaSystemAccountDetector")
+    static Predicate<Address> provideHederaSystemAccountDetector() {
+        // all addresses between 0-750 (inclusive) are treated as system accounts
+        // from the perspective of the EVM when executing Call, Balance, and SelfDestruct operations
+        return address ->
+                address.numberOfLeadingZeroBytes() >= 18 && Integer.compareUnsigned(address.getInt(16), 750) <= 0;
+    }
+
+    @Provides
+    @Singleton
+    @Named("StrictHederaSystemAccountDetector")
+    static Predicate<Address> provideStrictHederaSystemAccountDetector() {
+        // all addresses between 0-999 (inclusive) are treated as system accounts
+        // from the perspective of the EVM when executing ExtCode operations
+        return address ->
+                address.numberOfLeadingZeroBytes() >= 18 && Integer.compareUnsigned(address.getInt(16), 999) <= 0;
```

Comment:
I'm not sure why the 0-999 range is used here instead of 0-1000. I don't know the reason and I'm not certain that this is correct.

Reply:
Need to confirm this.